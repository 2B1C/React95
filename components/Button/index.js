import React from 'react';
import styled, { injectGlobal } from 'styled-components';

injectGlobal`
  @font-face {
    font-family: 'Pixelade';
    src: url('/Pixelade.woff2') format('woff2'),
        url('/Pixelade.woff') format('woff');
    font-weight: normal;
    font-style: normal;
  }

  body {
    background-color: #c3c7cb;
  }

  * {
    font-family: 'Pixelade';
  }
`

const Button = styled.button`
  background-color: #c3c7cb;
  padding: 5px 20px 6px;
  border: none;
  font-size: 20px;

  box-shadow:  inset 1px 1px 0px 1px #ffffff,
               inset 0 0 0 1px #868a8e,
               1px 1px 0 1px #000;

  &:disabled {
    color: #868a8e;
  }

  &:focus {
    outline: none;
    box-shadow: inset 1px 1px 0px 1px #ffffff,
                inset -1px -1px 0px 1px #868a8e,
                1px 1px 0 2px #000;
  }

  &:active {
    padding: 6px 20px 5px;

    outline: 1px dotted #000;
    outline-offset: -5px;
    outline-width: 2px;

    box-shadow: inset 0 0 0 1px #868a8e,
                0 0 0 1px #000;
  }
`;

export default Button;
