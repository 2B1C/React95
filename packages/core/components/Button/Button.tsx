import Btn from '../shared-style/Btn';

const Button = Btn;

Button.defaultProps = {
  children: 'Ok',
};

export default Button;
