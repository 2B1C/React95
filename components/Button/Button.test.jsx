import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';

describe('<Button />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      expect(renderer.create(<Button />).toJSON()).toMatchSnapshot();
    });
  });
});
