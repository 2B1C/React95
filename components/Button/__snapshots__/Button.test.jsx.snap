// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Snapshots should match snapshot 1`] = `
.c0 {
  background-color: #c3c7cb;
  padding: 4px 20px 4px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 1px #000;
}

.c0:disabled {
  color: #868a8e;
}

.c0:focus {
  outline: none;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -1px -1px 0px 1px #868a8e,1px 1px 0 2px #000;
}

.c0:active {
  padding: 5px 20px 4px;
  outline: 1px dotted #000;
  outline-offset: -5px;
  outline-width: 2px;
  box-shadow: inset 0 0 0 1px #868a8e,0 0 0 1px #000;
}

<button
  className="c0"
  onClick={[Function]}
  value="Ok"
>
  Ok
</button>
`;
